language/move-prover/spec-lang/src/env.rs

```diff
@@ -1940,3 +1976,58 @@ impl<'env> FunctionEnv<'env> {
         )
     }
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ConstantData {
+    pub loc: Loc,
+    pub name: Symbol,
+    pub type_: Type,
+    pub idx: ConstantPoolIndex,
+}
+
+#[derive(Debug, Clone)]
+pub struct ConstantEnv<'env> {
```

Comment:
I don't think you actually need a ConstantEnv. The XEnv idiom is used if there is something to lookup about, like e.g. it's a struct, and its fields. You can just have

```
pub fn get_constant(&self, id: ConstantId) -> &ConstantDecl
pub fn get_constants(&self) -> &[ConstantDecl]
```